R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,677 @@
+    "#' @param features The shape and moments generated by EBImage computeFeatures\n",
+    "#' @return A dataframe specifying the x, y, rx, ry, w, h, theta and text parameters of the ROIs\n",
+    "createROIs <- function(features) {\n",
+    "    rois <- data.frame(x = c(0), y = c(0), rx = c(0), ry = c(0), w = c(0), h = c(0), theta = c(0), text = c('remove'), stringsAsFactors = FALSE)\n",
+    "    for (index in 1:length(features[,1])) {\n",
+    "        x <- features[index,8]\n",
+    "        y <- features[index,9]\n",
+    "        r1 <- features[index,10]\n",
+    "        ecc <- features[index,11]\n",
+    "        r2 <- sqrt(r1^2 * (1 - ecc^2))\n",
+    "        theta <- features[index,12]\n",
+    "        rois <- rbind(rois, c(x, y, r1, r2, NA, NA, -theta, as.character(index)))\n",
+    "    }\n",
+    "    rois <- rois[-1,]\n",
+    "    rownames(rois) <- c()\n",
+    "    return(rois)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rois <- createROIs(features)\n",
+    "rois"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Save the ROIs back to OMERO"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# addROIs creates an ROI for each entry in the dataframe specified by the 'coords' parameter\n",
+    "addROIs(image, coords = rois)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open the image in the full viewer in OMERO.web and check the ROIs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Store the features table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Attach the whole 'features' dataframe to the image:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 'attachDataframe' directly attaches an R dataframe to an OMERO image (project, dataset, etc.)\n",
+    "# ('invisible' not strictly necessary, just suppresses some unneeded output)\n",
+    "invisible(attachDataframe(image, features, \"ROI features\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataframe is now attached to the image as an [HDF](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) table with file name 'ROI features'. You can download and open it with software like HDF Compass, load it into python scripts using 'h5py' library, etc. or simply load it directly as R dataframe again."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively attach it as CSV file:\n",
+    "\n",
+    "- Use R's built-in write.csv method to create a csv file\n",
+    "- And attach that to the image using attachFile function from the romero.gateway"
```

Comment:
I have in my csv only 2 lines, but I am having 7 ROIs in that image. Do you pre-select the objects to be evaluated ?

Reply:
Deleting the comment about the csv having only 2 lines - it is invalid, sorry, the number of lines is 7 as it should be